java/.gitignore

```diff
@@ -0,0 +1,4 @@
+bazel-bin
```

Comment:
This is a great idea!

How about adding a `.gitignore` on the top level of the repo, so that we do not track those folders in all nested workspaces?